Tests/AdyenTests/Adyen Tests/Components/Apple Pay/ApplePayComponentTests.swift

```diff
@@ -41,49 +41,55 @@ class ApplePayComponentTest: XCTestCase {
         UIApplication.shared.keyWindow!.rootViewController = emptyVC
     }
 
-    func testApplePayViewControllerIsDismissedFromInside() {
+    func testApplePayViewControllerShouldCallDelegateDidFail() {
         guard Available.iOS12 else { return }
-        let dummyExpectation = expectation(description: "Wait stop dismissing")
 
+        // This is nececery to give ApplePay time to dissapear from screen.
+        wait(for: .seconds(2))
+
+        let viewController = sut!.viewController
+        let onDidFailExpectation = expectation(description: "Wait for delegate call")
         mockDelegate.onDidFail = { error, component in
-            XCTFail("should not call didFail")
+            XCTAssertEqual(error as! ComponentError, ComponentError.cancelled)
+            onDidFailExpectation.fulfill()
         }
 
-        let viewController = sut.viewController
         UIApplication.shared.keyWindow!.rootViewController = emptyVC
-        UIApplication.shared.keyWindow!.rootViewController!.present(self.sut.viewController, animated: false)
+        UIApplication.shared.keyWindow!.rootViewController!.present(viewController, animated: false)
         
         wait(for: .seconds(1))
-        
-        self.sut.dismiss {
-            DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
-                dummyExpectation.fulfill()
-            }
-        }
+
+        self.sut.paymentAuthorizationViewControllerDidFinish(viewController as! PKPaymentAuthorizationViewController)
 
         waitForExpectations(timeout: 10)
+
         XCTAssertTrue(viewController !== self.sut.viewController)
     }
 
-    func testApplePayViewControllerShouldCallDelegateDidFail() {
+    func testApplePayViewControllerShouldCallFinalizeCompleation() {
```

Comment:
Typo :-), its ignored by the spell check CI job because its in test

Reply:
No, it is [passing on CI](
https://github.com/Adyen/adyen-ios/runs/5571626411?check_suite_focus=true#step:8:1001)